error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/CollectionRules.java

```diff
@@ -404,6 +405,19 @@ Optional<T> after(Queue<T> queue) {
     }
   }
 
+  /** Prefer calling {@link Collection#forEach} ()} over more contrived alternatives. */
+  static final class CollectionForEach<T> {
+    @BeforeTemplate
+    void before(Collection<T> collection, Consumer<T> consumer) {
+      collection.stream().forEach(consumer);
+    }
+
+    @AfterTemplate
+    void after(Collection<T> collection, Consumer<T> consumer) {
+      collection.forEach(consumer);
+    }
+  }
```

Comment:
The `forEach` methods accept a `Consumer<? super T>`. So we can define a more generic rule:

```suggestion
  static final class CollectionForEach<S, T extends S> {
    @BeforeTemplate
    void before(Collection<T> collection, Consumer<S> consumer) {
      collection.stream().forEach(consumer);
    }

    @AfterTemplate
    void after(Collection<T> collection, Consumer<S> consumer) {
      collection.forEach(consumer);
    }
  }
```